crates/forge/bin/cmd/coverage.rs

```diff
@@ -331,13 +320,13 @@ impl CoverageArgs {
             .set_coverage(true)
             .build(&root, output, env, evm_opts)?;
 
-        // Run tests
-        let filter = self.filter;
-        let (tx, rx) = channel::<(String, SuiteResult)>();
-        let handle = tokio::task::spawn_blocking(move || runner.test(&filter, tx));
+        let outcome = self
+            .test
+            .run_tests(runner, config.clone(), verbosity, &self.test.filter(&config))
+            .await?;
 
         // Add hit data to the coverage report
-        let data = rx.into_iter().flat_map(|(_, suite)| {
+        let data = outcome.results.into_iter().flat_map(|(_, suite)| {
```

Comment:
Note that this is now not ran in parallel with test execution, but that's probably fine.

Reply:
should be fine